### PR TITLE
feat: set up RTL

### PR DESCRIPTION
**Pull request by PavelKirvelTextkernel**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. Use imperative, present tense in your commit description ("change", not "changed" or "changes") without uppercases or period (.) at the end.

# Checklist
- [ ] The implementation has been manually tested and complies with Textkernel [browser support guidelines](https://textkernel.com/browser-support/)
- [ ] The implementation complies with [accessibility](CONTIRBUTING.md#accessibility) standards.
- [ ] The component has a [displayName](CONTRIBUTING.md#display-names) defined.
- [ ] The component comes with a detailed [PropTypes](CONTRIBUTING.md#component-props) (and defaultProps) definition.
- [ ] Component PropTypes are sufficiently described / documented.
- [ ] There is [a story](CONTRIBUTING.md#component-showcases) in Storybook.

### CODE REVIEW

**Comment by PavelKirvelTextkernel** [comment]
https://textkernel.atlassian.net/browse/ONEUI-365

File: package.json [hunk 1]
```diff
@@ -98,6 +98,9 @@
         "@storybook/react": "6.5.16",
         "@storybook/storybook-deployer": "2.8.16",
         "@storybook/testing-library": "0.0.13",
+        "@testing-library/react": "12.1.5",
+        "@testing-library/jest-dom": "5.16.5",
+        "@types/react-test-renderer": "18.0.0",
```
Comment (mukiienko) [inline package.json:103]: do we need [@testing-library/user-event](https://github.com/testing-library/user-event)?

File: .eslintrc.json [hunk 2]
```diff
@@ -58,7 +58,8 @@
         "@typescript-eslint/comma-dangle": ["error", "only-multiline"],
         "@typescript-eslint/explicit-module-boundary-types": "off",
         "@typescript-eslint/ban-ts-comment": "off",
-        "@typescript-eslint/ban-types": "off"
+        "@typescript-eslint/ban-types": "off",
+        "import/no-extraneous-dependencies": "off"
```
Comment (mukiienko) [inline .eslintrc.json:62]: tell me please why we need `import/no-extraneous-dependencies: off`?

File: src/components/Chip/__tests__/Chip.spec.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-import toJson from 'enzyme-to-json';
+import { create } from 'react-test-renderer';
 import { Chip } from '../Chip';
 
 describe('<Chip> that renders a pill shaped chip', () => {
     it('should render correctly', () => {
-        const wrapper = shallow(<Chip>some text</Chip>);
-        expect(toJson(wrapper)).toMatchSnapshot();
+        const wrapper = create(<Chip>some text</Chip>);
+        expect(wrapper).toMatchSnapshot();
```
Comment (mukiienko) [inline src/components/Chip/__tests__/Chip.spec.js:8]: do we need to convert to JSON by using toJSON() method?

Comment (PavelKirvelTextkernel) [reply src/components/Chip/__tests__/Chip.spec.js:8]: No, here we use new library it doesn't require it here you can check some example https://github.com/textkernel/oneui/pull/906/files

File: package.json (hunk 1 above)
Comment (PavelKirvelTextkernel) [reply package.json:103]: I used this MR as example http://tkgit-01.textkernel.net/searchbox/reach-ui/-/merge_requests/719/diffs we don't use it there. If we need it we can add when we will migrate all tests

File: .eslintrc.json (hunk 2 above)
Comment (PavelKirvelTextkernel) [reply .eslintrc.json:62]: Actually we need to use this versions:
"react": "18.2.0",
"react-dom": "18.2.0",
But the problem is  "react-virtualized": "~9.22.0"  get error react@"^15.3.0 || ^16.0.0-alpha"

https://github.com/bvaughn/react-virtualized/issues/1746

**Review by mukiienko — APPROVED** [review]

**Pull request merged**